mapper/validator.go

```diff
@@ -0,0 +1,39 @@
+package mapper
+
+import (
+	"github.com/Financial-Times/methode-content-placeholder-mapper/model"
+	"errors"
+	"net/url"
+)
+
+type CPHValidator interface {
+	Validate(mcp *model.MethodeContentPlaceholder) error
+}
+
+type defaultCPHValidator struct {
+}
+
+func NewDefaultCPHValidator() *defaultCPHValidator {
+	return &defaultCPHValidator{}
+}
+
+func (dcv *defaultCPHValidator) Validate(mcp *model.MethodeContentPlaceholder) error {
+	return dcv.validateHeadline(mcp.Body.LeadHeadline)
+}
+
+func (dcv *defaultCPHValidator) validateHeadline(headline model.LeadHeadline) error {
```

Comment:
Is it necessary for this function to have a receiver?

Reply:
I think it's okay. It's mostly due to encapsulations, so the validation function can be mocked in places where the validator is a dependency. In `mapper.go` you could inject a mocked validator.